data/cube/View.js

```diff
@@ -108,26 +107,23 @@ export class View {
     }
 
     getDimensionValues() {
-        const {_leafMap} = this,
-            fields = this._query,
+        const {_leaves} = this,
+            {fields} = this._query,
+            dims = pick(fields, f => f.isDimension),
             ret = [];
 
-        forEach(fields, f => {
-            if (f.isDimension) {
-                const vals = [];
-                _leafMap.forEach(rec => {
-                    const lVal = rec.get(f.name);
-                    if (lVal != null && !vals.includes(lVal)) {
-                        vals.push(lVal);
-                    }
-                });
-
-                ret.push({
-                    name: f.name,
-                    displayName: f.displayName,
-                    values: vals
```

Comment:
Don't think we're using this right now, but making `values` a Set would change the API, no? Obviously benefit in terms of code cleanup

Reply:
yes -- but we are essentially at a clean slate with the API -- the sencha upgrade is never going to be a dropin at this point.